lib/index.js

```diff
@@ -295,8 +310,162 @@ Segment.prototype.referrerId = function(query, ctx) {
   this.cookie('s:context.referrer', json.stringify(ad));
 };
 
+
 /**
- * Noop.
+ * retrieveCrossDomainId.
+ *
+ * @api private
+ * @param {function) callback => err, {crossDomainId, fromServer, timestamp}
+ */
+Segment.prototype.retrieveCrossDomainId = function(callback) {
+  if (!this.options.crossDomainIdServers) {
+    if (callback) {
+      callback('crossDomainId not enabled', null);
```

Comment:
See https://github.com/segment-integrations/analytics.js-integration-segmentio/pull/25 for both.